pkg/asset/installconfig/azure/session.go

```diff
@@ -128,18 +132,26 @@ func getCredentials(fs auth.FileSettings) (*Credentials, error) {
 		return nil, errors.New("could not retrieve clientId from auth file")
 	}
 	clientSecret := fs.Values[auth.ClientSecret]
-	if clientSecret == "" {
-		return nil, errors.New("could not retrieve clientSecret from auth file")
-	}
 	tenantID := fs.Values[auth.TenantID]
 	if tenantID == "" {
 		return nil, errors.New("could not retrieve tenantId from auth file")
 	}
+	clientCertificatePassword := fs.Values[auth.CertificatePassword]
+	clientCertificatePath := fs.Values[auth.CertificatePath]
+	if clientSecret == "" {
+		if clientCertificatePassword == "" || clientCertificatePath == "" {
```

Comment:
It's completely valid to have an empty `clientCertificatePassword`.  It's possible the certificate file is not encrypted at all. 

When we fetch an authorizer against the certificate, it calls the following function(s):
`c.Authorizer()` -> [calls](https://github.com/rna-afk/installer/blob/5c9a27ac82efb287c6e629bad2e1d7bcc97cdffb/vendor/github.com/Azure/go-autorest/autorest/azure/auth/auth.go#L622) `ServicePrincipalToken` -> [which attempts to decode the certificate file using the password](https://github.com/rna-afk/installer/blob/5c9a27ac82efb287c6e629bad2e1d7bcc97cdffb/vendor/github.com/Azure/go-autorest/autorest/azure/auth/auth.go#L613) -> `DecodeCertificatePfxData` [here](https://github.com/rna-afk/installer/blob/5c9a27ac82efb287c6e629bad2e1d7bcc97cdffb/terraform/providers/azurerm/vendor/github.com/Azure/go-autorest/autorest/adal/persist.go#L92) which has a comment at the top stating:
```
// If the private key is not password protected pass the empty string for password.
```